Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageLayoutOptionsResolver.swift

```diff
@@ -9,7 +9,11 @@ import StreamChat
 open class ChatMessageLayoutOptionsResolver {
     /// The maximum time interval between 2 consecutive messages sent by the same user to treat them as a single message group.
     public let maxTimeIntervalBetweenMessagesInGroup: TimeInterval
-
+    
+    // TODO: Propagate via `init` in v5.
+    /// The deleted messags visability. Is automatically populated from `ChatClient.config`.
+    public internal(set) var deletedMessagesVisibility: ChatClientConfig.DeletedMessageVisibility = .alwaysVisible
```

Comment:
Isn't it better to pass the whole ChatClientConfig? What if we need more stuff from it in the future? Then, we might keep adding new properties every time we need something from the chat client config. 

We kinda need to rethink this in v5, the resolver probably needs to have access to the Components config + the appearance config as well (Example: The date separators).

Can we add this to our V5 Document and link this PR as well? That would be nice! So that we don't forget about this (We might forget this TODO comment)

Reply:
Updated to pass the entire config ✅